hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/CacheCodecTemplate.java

```diff
@@ -405,4 +405,40 @@
     @Request(id = 32, retryable = true, response = ResponseMessageConst.LIST_ENTRY_PARTITION_UUID)
     @Since("1.4")
     Object assignAndGetUuids();
+
+
+    /**
+     * Performs the initial subscription to the cache event journal.
+     * This includes retrieving the event journal sequences of the
+     * oldest and newest event in the journal.
+     *
+     * @param name name of the cache
+     * @return the cache event journal subcription information
+     */
+    @Request(id = 33, retryable = true, response = ResponseMessageConst.EVENT_JOURNAL_INITIAL_SUBSCRIBER_STATE, partitionIdentifier = "partitionId")
+    @Since("1.5")
+    Object eventJournalSubscribe(String name);
+
+    /**
+     * Reads from the cache event journal in batches. You may specify the start sequence,
+     * the minimum required number of items in the response, the maximum number of items
+     * in the response, a predicate that the events should pass and a projection to
+     * apply to the events in the journal.
+     * If the event journal currently contains less events than {@code minSize}, the
+     * call will wait until it has sufficient items.
```

Comment:
No specific timeout for this method - it blocks the same way `com.hazelcast.client.impl.protocol.template.RingbufferCodecTemplate#readOne` blocks (and `com.hazelcast.client.impl.protocol.template.LockCodecTemplate#lock` for instance). I'm not sure if there is a limit on how long operations can be parked on the member side. I can check and add to the description if needed